onnxruntime/core/optimizer/reshape_fusion.cc

```diff
@@ -113,7 +111,8 @@ bool ReshapeFusion::Fuse_Subgraph1(Node& reshape, Graph& graph, const logging::L
     const Node& gather = edges[1]->GetNode();
     const Node& shape = edges[2]->GetNode();
 
-    if (graph_utils::GetInputNode(shape, 0) != p_root) {
+    const NodeArg& shape_input = *(shape.InputDefs()[0]);
+    if (shape_input.Name() != root_input.Name()) {
```

Comment:
Can you compare pointers instead?

Reply:
That relies on assumption that graph will not allocate NodeArg twice for same name. Comparing  by name does not rely on such implementation detail of graph.